### ISSUE TITLE
Fluent integration

### ISSUE THREAD

**Issue by brettt89** [brettt89/silverstripe-versioned-gc#3]
When using Fluent, version records are expanded to include locale information and duplication. Previous iteration of this code (https://github.com/silverstripe/silverstripe-versioned/pull/335) had Fluent integration built in, however integrating this would require the fluent module.

So instead, Extensible was used on `VersionedCollector` so that an Extension can be used to hook in Fluent modifications of queries for this collector.

This Issue was created to discuss how/where we should have fluent integration

E.g.
 - Within Fluent module
 - In its own Module (would be small, maybe just an extension + yaml).
 - Within this module (Perhaps we would need to think about composer requirements as we dont want to force usage of Fluent on people who just want Versioned-GC without this code)

**Comment by mfendeksilverstripe** [comment]
Given the scope of the Fluent related code is small, maybe this is an extension in this model which is NOT applied by default. Fluent can be suggested as a recommended module via composer.

**Comment by brettt89** [comment]
> Given the scope of the Fluent related code is small, maybe this is an extension in this model which is NOT applied by default. Fluent can be suggested as a recommended module via composer.

Yeah that was my thinking as well. We can adjust `Testing` instructions to manually require `fluent` module when testing this extension.

@mfendeksilverstripe Using this theory, would it make sense to apply this same logic for Versioned / ChangeSet collectors and add them to the base Garbage Collection Service module?

(E.g. Add the Collectors defined in this module to the BaseModule, but NOT applied by default?). This way people only need to require a single module and it contains all default collectors and they can enable/configure the ones they want?

**Comment by mfendeksilverstripe** [comment]
Sounds good @brettt89 , keeping the code in one module makes sense in this context 👍

**Comment by brettt89** [comment]
> Sounds good @brettt89 , keeping the code in one module makes sense in this context 👍

Okay will look to merge them together into a single repository. Appreciate the feedback.
